test/signup.test.js

```diff
@@ -0,0 +1,168 @@
+        expect(response.status).toBe(validationErrorStatus);
+      });
+  });
+
+  test('Create user with used email', () => {
+    factory.define('user', Users, buildOptions => ({
+      first_name: buildOptions.firstName,
+      last_name: buildOptions.lastName,
+      email: buildOptions.email,
+      password: buildOptions.password
+    }));
+    expect.assertions(1);
+    return factory
+      .create('user', { firstName, lastName, email: correctEmail, password: correctPassword })
+      .then(() =>
+        request(app)
+          .post('/users')
+          .send({
+            first_name: firstName,
+            last_name: lastName,
+            email: correctEmail,
+            password: correctPassword
+          })
+      )
+      .then(response => {
+        expect(response.status).toBe(validationErrorStatus);
+      });
+  });
+});
+
+describe('Test missing parameters', () => {
+  test('Create user without password', () => {
+    expect.assertions(1);
+    return request(app)
+      .post('/users')
+      .send({
+        first_name: firstName,
+        last_name: lastName,
+        email: correctEmail
+      })
+      .then(response => {
+        expect(response.status).toBe(validationErrorStatus);
+      });
+  });
+
+  test('Create user without email', () => {
+    expect.assertions(1);
+    return request(app)
+      .post('/users')
+      .send({
+        first_name: firstName,
+        last_name: lastName,
+        password: correctPassword
+      })
+      .then(response => {
+        expect(response.status).toBe(validationErrorStatus);
+      });
+  });
+
+  test('Create user without first name', () => {
+    expect.assertions(1);
+    return request(app)
+      .post('/users')
+      .send({
+        last_name: lastName,
+        email: correctEmail,
+        password: correctPassword
+      })
+      .then(response => {
+        expect(response.status).toBe(validationErrorStatus);
+      });
+  });
+
+  test('Create user without last name', () => {
+    expect.assertions(1);
+    return request(app)
+      .post('/users')
+      .send({
+        first_name: firstName,
+        email: correctEmail,
+        password: correctPassword
+      })
+      .then(response => {
+        expect(response.status).toBe(validationErrorStatus);
+      });
+  });
+
+  test('Create user without any parameter', () => {
+    expect.assertions(1);
+    return request(app)
+      .post('/users')
+      .then(response => {
+        expect(response.status).toBe(validationErrorStatus);
+      });
+  });
+});
+
+describe('Creat user correctly', () => {
+  test('Create user with all paramaters set up correctly', () => {
+    expect.assertions(1);
+    return request(app)
+      .post('/users')
+      .send({
+        first_name: firstName,
+        last_name: lastName,
+        email: correctEmail,
+        password: correctPassword
+      })
+      .then(response => {
+        expect(response.status).toBe(createdCorrectlyStatus);
```

Comment:
You should also assert if the params are actually set correctly in the database. Good unit testing practice indicates that you should make another test for these assertions, but we usually don't require it.